### ISSUE TITLE
Document format of input datasets

### ISSUE THREAD

**Issue by pmcharrison** [tuomaseerola/onsetsync#12]
@tuomaseerola I had a student try to run some analyses using the package and they struggled to work out what kind of format is required when providing one's own dataset. There is some information on this in the paper but I think it would be helpful to document it in the package itself too, ideally for all functions that ingest such datasets (roxygen has functionality for inheriting documentation so you don't have to copy and paste each time). Ideally this documentation would explain which columns are mandatory and which are optional. I also found the definition of Isochronous.SD.Time a bit confusing and wondered whether it could be explained a bit clearer, e.g. through the use of an example?

**Comment by pmcharrison** [comment]
From @tuomaseerola:

> I do recognise the issue myself, the existing examples are way above the "minimal required" information and explaining the required representations (columns of data) would be helpful. I have not thought through whether every function then would have to be checked and signposted for the specific requirements, it probably needs input check routines and documentation for each function, and turns easily into changing the the input to S3 classes but if the minimal change would be to add documentation to the package using a vignette (e.g. "input-representation"), this could be an easy task.

and my reply:

Hi @tuomaseerola, the vignette would indeed be a nice solution and you should do that if you think it's most efficient. The other (near-?) minimal solution would be to go to `plot_by_beat.R`, and update this line:

```
#' @param df data frame to be processed
```

to specify complete information about the required columns, then in every other function that uses such a dataset you delete that line and instead write 

```
#' @inheritParams plot_by_beat
```

**Comment by tuomaseerola** [comment]
Hi @pmcharrison. Very cool shortcut from roxygen2, thank you. I opted to document the function inputs explicitly (as the inheritParams only worked for some functions), and all functions now list the required inputs and default parameters in the documentation. I also created a [minimal_representation vignette/article](https://tuomaseerola.github.io/onsetsync/articles/minimal-representation.html). This also led to me to clarify the reference beat naming and calculation, which was initially inherited from the IEMP corpus (the awkward "Isochronous.SD.Time", which will still be evident in some of the documentation when annotated materials are used). If the reference timing is calculated from several instruments, it will be mean onset time and will be labelled as "Mean.Time". If the reference time is inferred from an external reference (Cycle timings from annotations), these are then isochronous time intervals for the cycles and labelled as "Iso.Time".  Documentation and website reflects these updates (this is now 0.5.1 version and the changes are in the R1 branch).
